packages/plugin-ma-tracking/src/ma-tracking.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  AdditionalBrowserOptions,
+  BaseEvent,
+  BeforePlugin,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import type { AmplitudeBrowser } from '@amplitude/analytics-browser';
+import { AdvancedCampaignTracker } from './advanced-campaign-tracker';
+
+export class MATrackingPlugin implements BeforePlugin {
+  name = 'MATracking';
+  type = PluginType.BEFORE as const;
+
+  constructor(private readonly instance: AmplitudeBrowser, private readonly options: AdditionalBrowserOptions = {}) {}
+
+  async setup(config: BrowserConfig): Promise<undefined> {
+    const attributionConfig = this.options.attribution ?? {};
+    const pageTrackingConfig =
+      this.options.trackPageViews ??
+      (attributionConfig.trackPageViews && {
+        filter: 'onAttribution',
+      });
+
+    const instance = this.instance;
+    // Share cookie storage with user session storage
+    const storage = config.cookieStorage as unknown as Storage<Campaign>;
+
+    const campaignTracker = new AdvancedCampaignTracker(config.apiKey, {
+      ...attributionConfig,
+      trackPageViews: false,
+      track: /* istanbul ignore next */ () => Promise.resolve(),
+      onNewCampaign: /* istanbul ignore next */ () => () => undefined,
+      storage,
+    });
+
+    // Disable "runAttributionStrategy" function from instance
+    instance.runAttributionStrategy = () => Promise.resolve();
```

Comment:
Lets avoid doing this anywhere please

Reply:
Any suggestion of preventing calling `runAttributionStrategy` from core SDK?

Reply:
shared here: https://amplitude.slack.com/archives/D035W3B4CJC/p1661982534275369